src/rt/tracegc.d

```diff
@@ -104,14 +119,21 @@ private string generateTraceWrappers()
         static if (name.length > 3 && name[0..3] == "_d_")
         {
             mixin("alias Declaration = " ~ name ~ ";");
-            code ~= generateWrapper!Declaration();
+            code ~= generateWrapper!(Declaration, ParamPos.front)();
+        }
+        static if (name.length > 3 && name[0..3] == "gc_")
+        {
+            mixin("alias Declaration = " ~ name ~ ";");
+            code ~= generateWrapper!(Declaration, ParamPos.back)();
```

Comment:
For minimal intrusiveness, I decided to tack on the tracing parameters as default parameters at the end of the parameter list. But that means we can't use the parameter order here for the `gc_xxx` functions, since `core.memory.GC` forwards them here via `pragma(mangle)`, so there's no way to reorder the parameters nicely.  For that reason, I decided it's easier to change the wrapper for `gc_xxx` to take the file, line, funcname parameters at the end of the parameter list instead, then we can just continue to use `pragma(mangle)` to forward the GC calls.